src/modules/Elsa.MongoDB/Common/MongoStore.cs

```diff
@@ -0,0 +1,170 @@
+using System.Linq.Expressions;
+using Elsa.MongoDB.Extensions;
+using JetBrains.Annotations;
+using MongoDB.Driver;
+using MongoDB.Driver.Linq;
+
+namespace Elsa.MongoDB.Common;
+
+/// <summary>
+/// A generic repository class around MongoDb for accessing documents.
+/// </summary>
+/// <typeparam name="TDocument">The type of the document.</typeparam>
+[PublicAPI]
+public class MongoStore<TDocument> where TDocument : class
```

Comment:
I think that, since the name is "MongoDB", a better name for this class would be "MongoDBStore" (or "MongoDbStore", depending on the conclusion of one of my previous comments).